### PR TITLE
M3 fix templating helper

### PR DESCRIPTION
**Pull request by hluchas**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/8284
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
When TemplatingHelper was used via CLI, it went to fatal error.
Used PR https://github.com/mautic/mautic/pull/8276

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Use local SMTP transport
2. Manually enable integration in column `mautic_plugin_integration_settings.is_publisheed` by swithing to `1`
2. Ensure that you have at least 1 email to be send in one campaign
3. Call `bin/console mautic:campaigns:trigger -e dev` via cli
4. You'll get
```
In TemplatingHelper.php line 48:
                                                                                                                     
  [Symfony\Component\Debug\Exception\UndefinedMethodException]                                                       
  Attempted to call an undefined method named "enterScope" of class "ContainerLn4vclt\appDevDebugProjectContainer".  
                                                                                                                     

Exception trace:
  at /Users/lukas.drahy/dev/community-fork/app/bundles/CoreBundle/Helper/TemplatingHelper.php:48
 Mautic\CoreBundle\Helper\TemplatingHelper->getTemplating() at /Users/lukas.drahy/dev/community-fork/plugins/MauticCitrixBundle/EventListener/EmailSubscriber.php:204
 MauticPlugin\MauticCitrixBundle\EventListener\EmailSubscriber->decodeTokens() at /Users/lukas.drahy/dev/community-fork/plugins/MauticCitrixBundle/EventListener/EmailSubscriber.php:149
```

#### Steps to test this PR:
1. Same as reproduction steps

### CODE REVIEW

**Comment by escopecz** [comment]
I cannot replicate:
```
bin/console mautic:campaigns:trigger -e dev -v
Triggering events for campaign 581
Triggering events for newly added contacts
5 total events(s) to be processed in batches of 100 contacts
 5/5 [============================] 100%
5 total events were executed
0 total events were scheduled
...
```

I can see that the error originates from the Citrix bundle. Which one from the GoTo* plugins do you have enabled?

File: app/bundles/CoreBundle/Helper/TemplatingHelper.php [hunk 1]
```diff
@@ -39,21 +40,24 @@ public function __construct(Kernel $kernel)
...
+     *
+     * @throws \Exception
      */
     public function getTemplating()
     {
         if (defined('IN_MAUTIC_CONSOLE')) {
-            //enter the request scope in order to be use the templating.helper.assets service
-            $this->container->enterScope('request');
-            $this->container->set('request', new Request(), 'request');
+            // Use the request scope in order to be the templating.helper.assets service used
```
Comment (escopecz) [inline app/bundles/CoreBundle/Helper/TemplatingHelper.php]: This comment is out of date as the `request` scope was removed in Symfony 3:

https://symfony.com/blog/new-in-symfony-2-8-deprecating-scopes-and-introducing-shared-services

I checked the `templating.helper.assets` service and I don't see any use of Request there. Where does it fail when you remove this condition altogether?

File: app/bundles/CoreBundle/Helper/TemplatingHelper.php [hunk 2]
```diff
@@ -39,21 +40,24 @@ public function __construct(Kernel $kernel)
...
+            $this->container->set('request', new Request());
         }
 
         return $this->container->get('templating');
     }
 
     /**
      * @return TemplateNameParser
+     *
+     * @throws \Exception
```
Comment (escopecz) [inline app/bundles/CoreBundle/Helper/TemplatingHelper.php]: Why was this added? I can't find where it throws an exception.

File: app/bundles/CoreBundle/Helper/TemplatingHelper.php (hunk 1 above)
Comment (hluchas) [reply app/bundles/CoreBundle/Helper/TemplatingHelper.php]: I can't reproduce this too without manually enabled integration in DB. GoToWebinar plugin was enabled without credentials. I checked iif I can do this in UI and it's not possible. But this fixes calls of deprecated methods.

When I enable integration in DB and remove condition it works. But this is not usual settings of app.
It had sense I believe if it is there. Request/stack is not available in container when called via CLI. I'll try to find why

**Comment by hluchas** [comment]
I can't reproduce this too. GoToWebinar plugin was enabled without credentials. I checked iif I can do this manually and it's not possible. But this fixes calls of deprecated methods.

> I cannot replicate:
> 
> ```
> bin/console mautic:campaigns:trigger -e dev -v
> Triggering events for campaign 581
> Triggering events for newly added contacts
> 5 total events(s) to be processed in batches of 100 contacts
>  5/5 [============================] 100%
> 5 total events were executed
> 0 total events were scheduled
> ...
> ```
> 
> I can see that the error originates from the Citrix bundle. Which one from the GoTo* plugins do you have enabled?
I can't reproduce this too without manually enabled integration in DB. GoToWebinar plugin was enabled without credentials. I checked iif I can do this in UI and it's not possible. But this fixes calls of deprecated methods.

When I enable integration in DB and remove condition it works. But this is not usual settings of app.
It had sense I believe. Request/stack is not available in container when called via CLI. I'll try to find why

File: app/bundles/CoreBundle/Helper/TemplatingHelper.php (hunk 1 above)
Comment (hluchas) [reply app/bundles/CoreBundle/Helper/TemplatingHelper.php]: Not found. Removed.

File: app/bundles/CoreBundle/Helper/TemplatingHelper.php (hunk 2 above)
Comment (hluchas) [reply app/bundles/CoreBundle/Helper/TemplatingHelper.php]: It is autodetected by code inspection in PhpStrom. I tried to search for this too  and not found. Removed

**Comment by hluchas** [comment]
Removed condition. Updated reproduction steps

**Pull request merged**